### ISSUE TITLE
Using S3 for artifacts is slow

### ISSUE THREAD

**Issue by eddycharly** [tektoncd/pipeline#1535]
Using a S3 bucket for artifacts storage is slow when there is a lot of files to copy.

As it is an automatic process managed by tekton itself, could we automatically zip the artifact before uploading and unzip it after downloading ?

It should be transparent from the user point of view and should improve performance a lot.
Maybe we can add a flag in the config map to enable/disable the zip/unzip behaviour.

I can submit a pull request for this if it's ok.

**Comment by vdemeester** [comment]
@eddycharly what are you using s3 with ? (I am guessing a storage PipelineResource)

/cc @iancoffey @dlorenc 
/kind question

**Comment by iancoffey** [comment]
@eddycharly FWIW - compression/decompression is definitely always something I envisioned this resource being able to optionally just handle entirely as you described, and I think it would be a great addition 👍

**Comment by mmpei** [comment]
this will speed a lot for resource input and output. it will also reduce the files uploading to storage and makes managing of artifacts easier.

**Comment by eddycharly** [comment]
Sorry for the late reply. I have almost stopped using outputs in my pipelines in favor of manually uploading/downloading the artifacts directly in my tasks. 
That’s not ideal for reusing a task across different pipelines but i couldn’t find a better way to use s3 as a storage with decent performance and enough freedom when creating pipelines. 

I will have another look as I hope it will improve in the next versions of Tekton.

About the automatic compression/décompression, the last time I looked at the code it was really easy to add. I will try to make a pr when I have some time. 

What I would really love to see is a way to define how data flows from one task to another inside a pipeline without having to bother about storage at all (all I should have to do is configure Tekton to use a bucket of my choice). 
When I tried, I found myself declaring git resources as output everywhere, renaming was not possible, and in the end it looked quite not natural and too restrictive.
